script/make-tra

```diff
@@ -58,41 +50,46 @@ check-command dd
 check-command tr
 check-command zip
 
-# validate existence of firmware images
-[ -e "$TANGARA_BIN" ] || die "firmware image not found: $TANGARA_BIN"
-[ -e "$LUA_BIN" ] || die "lua image not found: $LUA_BIN"
+if ! [ -f "$FIRMWARE_REPO/build/tangara.bin" ]; then
+    die "Firmware not found. Use --fwrepo to specify directory of tangara-fw repository containing built firmware"
+fi
 
 # read version out of firmware image
-[ -n "$VERSION" ] || VERSION="$(dd if="$TANGARA_BIN" bs=16 count=1 skip=3 status=none | tr -d '\000')" || die "reading version from firmware image"
+[ -n "$VERSION" ] || VERSION="$(dd if="$FIRMWARE_REPO/build/tangara.bin" bs=16 count=1 skip=3 status=none | tr -d '\000')" || die "reading version from firmware image"
 [ -n "$VERSION" ] || die "no version in firmware image"
 echo "$(tput setaf 4 bold)Found firmware version:$(tput sgr0) $(tput bold)${VERSION}$(tput sgr0)"
+OUTPUT_FILENAME="tangarafw-v$VERSION.tra"
 
 # create staging dir for zip
 TMP_DIR="/tmp/make-tra-$$"
 mkdir -p "$TMP_DIR" || die "creating staging dir: $TMP_DIR"
 
 # write the needful
-cp "$TANGARA_BIN" "$TMP_DIR/tangara.bin" || die "copying firmware image to staging dir"
-cp "$LUA_BIN" "$TMP_DIR/lua.bin" || die "copying lua image to staging dir"
+cp "$FIRMWARE_REPO/build/bootloader/bootloader.bin" "$FIRMWARE_REPO/build/partition_table/partition-table.bin" "$FIRMWARE_REPO/build/ota_data_initial.bin" "$FIRMWARE_REPO/build/tangara.bin" "$FIRMWARE_REPO/tools/collate/Generic.LC_COLLATE" "$FIRMWARE_REPO/build/lua.bin" "$FIRMWARE_REPO/build/repl.bin" "$TMP_DIR" || die "copying firmware images to staging dir"
 cat > "$TMP_DIR/tangaraflash.json" <<JSON || die "writing json manifest to staging dir"
     {
         "version": 0,
         "data": {
             "firmware": {
                 "version": "$VERSION",
                 "images": [
+                    { "addr": 4096, "name": "bootloader.bin" },
+                    { "addr": 32768, "name": "partition-table.bin" },
+                    { "addr": 53248, "name": "ota_data_initial.bin" },
                     { "addr": 65536, "name": "tangara.bin" },
-                    { "addr": 11599872, "name": "lua.bin" }
+                    { "addr": 8454144, "name": "Generic.LC_COLLATE" },
```

Comment:
We should avoid overwriting the collation partition during routine firmware updates, since it's very reasonable for someone to want to flash alternate collation data for their specific locale

Reply:
That would be reasonable, but is it documented? I wouldn't expect anyone to actually do that without documentation.

I'm unsure this will be as robust for recovering from bad/interrupted flashes if a partition is excluded.